tests/oper.rs

```diff
@@ -271,6 +271,45 @@ fn fold_and_sum() {
     }
 }
 
+#[test]
+fn fold_and_prod() {
```

Comment:
This can be renamed to just `prod` or `scalar_prod` after removing the block I pointed out.